File: java/414.java
class Solution {
    long first = Long.MIN_VALUE;
    long second = Long.MIN_VALUE;
    long third = Long.MIN_VALUE;
    
    public int thirdMax(int[] nums) {
        int l = nums.length;
        
        if(l < 3){
            int i = nums[0];
            for(int j:nums){
                i = Math.max(i,j);
            }
            return i;
        }
        for(int i:nums){
            if(i > first){
                third = second;
                second = first;
                first = i;
                
            }else if( i < first && i > second ){
                third = second;
                second = i;
            }else if(i < second && i > third){
                third = i;
            }
        }
        return third == Long.MIN_VALUE ? (int)first:(int)third;
    }
}

File: go/414.go
const INT_MAX = int(^uint(0) >> 1)
const INT_MIN = ^INT_MAX

func thirdMax(nums []int) int {
	first, second, third := INT_MIN, INT_MIN, INT_MIN
	for _, v := range nums {
		if v > first {
			third = second
			second = first
			first = v
		} else if v < first && v > second {
			third = second
			second = v
		} else if v > third && v < second {
			third = v
		}
	}
	if third == INT_MIN {
		return first
	}
	return third
}